sc/src/main/java/uk/ac/ebi/atlas/tsne/TSnePlotCollator.java

```diff
@@ -27,7 +25,30 @@ public TSnePlotCollator(TSnePlotDao tSnePlotDao, ClusterDao clusterDao, GeneExpr
         this.geneExpressionDao = geneExpressionDao;
     }
 
-    public TreeMultimap<Integer, TSnePoint> getTSnePlotWithClusters(String experimentAccession, int k) {
+    public TreeMultimap<Integer, TSnePoint> getTsnePlotWithClustersAndExpression(String experimentAccession,
+                                                                                 int k,
+                                                                                 String geneId) {
+        TreeMultimap<Integer, TSnePoint> clusters = this.getTSnePlotWithClusters(experimentAccession, k);
+        Map<String, Double> expression = geneExpressionDao.fetchGeneExpression(experimentAccession, geneId);
+
+        TreeMultimap<Integer, TSnePoint> clusterPointsWithExpression = TreeMultimap.create();
+        clusters.entries().forEach(
```

Comment:
Doesn't Java have `mapValues` ?

Reply:
No, it doesn’t AFAIK. You can `collect(toMap(...))` but that returns a map, not a multimap.